src/main/java/org/broadinstitute/hellbender/utils/collections/IntervalsSkipList.java

```diff
@@ -0,0 +1,130 @@
+package org.broadinstitute.hellbender.utils.collections;
+
+import com.google.common.collect.Lists;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+
+/**
+ * Holds many intervals, with an efficient operation to get
+ * intervals that overlap a given query interval.
+ *
+ * This version assumes that all the intervals lie on the same contig.
+ */
+public final class IntervalsSkipList<T extends Locatable> implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    // approx number of buckets we're aiming for.
+    private final int NUMBUCKETS = 1000;
+    // each bucket contains 2**SHIFT entries.
+    private final int SHIFT;
+    // reach: bucket# -> how far that bucket reaches.
+    // e.g. bucket 0 contains the first 2**SHIFT locatables. reach[0] is the max over their .getEnd()
+    //      reach[x] is the max over the .getEnd for that bucket and all the ones before it.
+    private int[] reach;
+
+    // input intervals, sorted by start location
+    private ArrayList<T> vs;
```

Comment:
always use interface (List) in declarations. The only place to put 'ArrayList' is in 'new ArrayList'